src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs

```diff
@@ -95,23 +102,33 @@ public class WebSocketsTransport : ITransport
                 {
                     using (message)
                     {
-                        await _webSocket.SendAsync(new ArraySegment<byte>(message.Payload.Buffer.ToArray()),
+                        try
+                        {
+                            await _webSocket.SendAsync(new ArraySegment<byte>(message.Payload.Buffer.ToArray()),
                             message.MessageFormat == Format.Text ? WebSocketMessageType.Text : WebSocketMessageType.Binary, true,
                             cancellationToken);
+                        } catch(OperationCanceledException ex)
+                        {
+                            _logger?.LogError("The WebSocket operation was cancelled");
```

Comment:
Is cancellationToken is guaranteed to be canceled when you get this exception? If not, you should break from the loop.